config/thisroot.sh

```diff
@@ -85,49 +110,51 @@ set_environment()
       return 1
    fi
 
+   local version=$1
+
    if [ -z "${PATH}" ]; then
       PATH=@bindir@; export PATH
    else
       PATH=@bindir@:$PATH; export PATH
    fi
 
    if [ -z "${LD_LIBRARY_PATH}" ]; then
-      LD_LIBRARY_PATH=@libdir@
+      LD_LIBRARY_PATH=@libdir@:@libdir@/python${version}
```

Comment:
Before this if, add a variable `exp_pyroot` that is set to true if `@libdir@/python${version}` exists. Then use that variable here and in the subsequent variable settings to add the python directory only if it exists, in order not to pollute the environment when building with current PyROOT.